### ISSUE TITLE
Double encoded query string

### ISSUE THREAD

**Issue by yaru22** [rackt/react-router#668]
Thanks for this awesome library by the way =)

If I transition to a route (let's say `/search`) with query param, for example, `{ q: "hello world" }`, the resulting URI becomes "/search?q=hello%2520world" whereas I expect it to be "/search?q=hello%20world".

Because of this, if I run `this.getQuery()` in the route matched component, I get `{ q: "hello%20world" }` as opposed to `{ q: "hello world" }`.

Is there a workaround I can use before the bug is fixed?

**Comment by mjackson** [comment]
@yaru22 This was fixed in fe5ec395f98e1d063539dd2211b590dd0840de59. It will be present in the 0.12 release.

**Comment by kuldeepdaftary** [comment]
So I'm trying to run following cases 

**Case 1**
```
const queryParam = {
  search: encodeURIComponent(searchTerm)
}
 this.props.goToRoute({
   pathname: this.props.locationBeforeTransitions.pathname,
   query: queryParam
  }); 
```
```bindActionCreators({ goToRoute: routerActions.replace }, dispatch)```

**Input**: "Hello World"
**Expected**: "Hello%20World"
**Actual**: "Hello%2520World"

Its Encoding query params twice

=======

**Case 2**
```
const queryParam = {
  search: encodeURIComponent(searchTerm)
}
 
this.props.goToRoute(`${this.props.locationBeforeTransitions.pathname}?search=${queryParam.search}`);
```

```bindActionCreators({ goToRoute: routerActions.replace }, dispatch)```

**Input**: "Hello World"
**Expected**: "Hello%20World"
**Actual**: "Hello%20World"

========

So Case 1 is encoding the queryParam twice where as Case 2 is doing it once and as expected.

Reason to encode queryParam is I need space to be encode as %20 rather than + and by default query param spaces are encode using space

Any other way can I change the default encoding ?
